### PR TITLE
Backups: Convert to feature checks

### PR DESCRIPTION
**Pull request by obenland**
#### Changes proposed in this Pull Request:

* No functional changes, everything should work as before
* Moves from plan checks to feature checks to determine backups support.

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Set up a jurassic ninja site with Jetpack beta and select this branch.
* Upgrade to a daily backup plan and save the credentials.
* Verify the Backups box in At a Glance looks like this: 
image

* Upgrade to a real time backup plan (like Complete)
* Verify the Backups box in At a Glance looks like this: 
image

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/at-a-glance/backups.jsx [hunk 1]
```diff
@@ -215,9 +208,9 @@ class DashBackups extends Component {
 			} );
 		}
 
-		if ( ! isEmpty( sitePlan ) ) {
+		if ( ! isFetchingSite ) {
 			// If site has a paid plan
-			if ( 'jetpack_free' !== get( sitePlan, 'product_slug', 'jetpack_free' ) ) {
+			if ( hasVaultPressBackups ) {
```
Comment (obenland) [inline projects/plugins/jetpack/_inc/client/at-a-glance/backups.jsx:213]: @vianasw The new VaultPress Backups feature is currently associated with JP premium and business plans (2c4c3-pb/#plain), which doesn't seem to line up with this conditional of any paid plan. Should it be premium and higher plans (2c4c4-pb/#plain) or truly any paid plan?

(If you're not the best person to ask, could you point me to who would be?)

Comment (ebinnion) [reply projects/plugins/jetpack/_inc/client/at-a-glance/backups.jsx:213]: I think personal and higher is a better fit as Personal (a now legacy plan) also had backups.

File: projects/plugins/jetpack/_inc/client/at-a-glance/backups.jsx [hunk 2]
```diff
@@ -294,10 +287,12 @@ class DashBackups extends Component {
 						) }
 					</React.Fragment>
 				);
-			case 'active':
-				const message = containsBackupRealtime( planClass )
-					? __( 'We are backing up your site in real-time.', 'jetpack' )
-					: __( 'We are backing up your site daily.', 'jetpack' );
+			case 'active': {
+				/* Avoid ternary as code minification will break translation function. :( */
```
Comment (ebinnion) [inline projects/plugins/jetpack/_inc/client/at-a-glance/backups.jsx:291]: Is this p2'd or described anywhere?

Comment (obenland) [reply projects/plugins/jetpack/_inc/client/at-a-glance/backups.jsx:291]: I've not looked but it seems like I'm not the first one to have come across this
https://github.com/Automattic/jetpack/blob/a860290cfdf9978014e9ba2e34363cd9df6d4e88/projects/plugins/jetpack/_inc/client/components/support-card/index.jsx#L92

Comment (obenland) [reply projects/plugins/jetpack/_inc/client/at-a-glance/backups.jsx:291]: It looks like it's a documented bug: https://github.com/Automattic/jetpack/tree/master/projects/js-packages/i18n-check-webpack-plugin#conditional-function-call-compaction

File: projects/plugins/jetpack/_inc/client/at-a-glance/backups.jsx (hunk 1 above)
Comment (ebinnion) [reply projects/plugins/jetpack/_inc/client/at-a-glance/backups.jsx:213]: Looking further here, I'm not sure that there needs to be any differentiation between VaultPress backups and Jetpack Backups. 

The issue as I understand it is more of a technical one, not one driven by whether a plan has been purchased or not.

That being said, there should be no issue keeping the VaultPress Backups feature check.

 @vianasw @jwebbdev - Will one of you confirm this please?

Comment (jwebbdev) [reply projects/plugins/jetpack/_inc/client/at-a-glance/backups.jsx:213]: Agreed, no distinction is required between VaultPress and Jetpack for Jetpack plans.  As far as I know all paid **bundle** plans include backup, then there are also the individual backup product plans.

**Review by mmtr — APPROVED** [review]
Looks good! Nice line diff count 🙂

**Pull request merged**